compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -706,6 +749,13 @@ pub fn compile_declarative_macro(
                 feature_err(sess, sym::macro_attr, span, "`macro_rules!` derives are unstable")
                     .emit();
             }
+            if unsafe_rule {
+                let err = sess.dcx().struct_span_err(
+                    unsafe_keyword_span,
+                    "`unsafe` is only supported on `attr` rules",
+                );
+                return dummy_syn_ext(err.emit());
```

Comment:
Nit: no need to reset to dummy, the macro itself can still be used for recovery.
Same for fn-like rules.

Reply:
Done.